File: swapi-server/src/main/java/de/lkaktus/entities/VehicleDTO.java
package de.lkaktus.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record VehicleDTO(String consumables,
                         @JsonProperty("cost_in_credits") String constInCredits,
                         String created,
                         String crew,
                         String edited,
                         String length,
                         String manufacturer,
                         @JsonProperty("max_atmosphering_speed") String maxAtmospheringSpeed,
                         String model,
                         String name,
                         String passengers,
                         @JsonProperty("vehicle_class") String vehicleClass,
                         String url,
                         String [] films,
                         SearchItemDTO [] filmItems,
                         String [] pilots,
                         SearchItemDTO [] pilotItems
) {

}

File: swapi-angular/src/app/api/models/VehicleDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { SearchItemDTO } from './SearchItemDTO';

export type VehicleDTO = {
    cost_in_credits?: string;
    max_atmosphering_speed?: string;
    vehicle_class?: string;
    consumables?: string;
    created?: string;
    crew?: string;
    edited?: string;
    length?: string;
    manufacturer?: string;
    model?: string;
    name?: string;
    passengers?: string;
    url?: string;
    films?: Array<string>;
    filmItems?: Array<SearchItemDTO>;
    pilots?: Array<string>;
    pilotItems?: Array<SearchItemDTO>;
};

